src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -36,7 +36,24 @@ public Task<Geoposition> GetGeopositionAsync()
 		public Task<Geoposition> GetGeopositionAsync(TimeSpan maximumAge, TimeSpan timeout)
 			=> GetGeopositionAsync();
 
-		public static async Task<GeolocationAccessStatus> RequestAccessAsync() => GeolocationAccessStatus.Allowed;
+		public static async Task<GeolocationAccessStatus> RequestAccessAsync()
+		{
+			// below 6.0 (API 23), permission are granted
+			if (Android.OS.Build.VERSION.SdkInt < Android.OS.BuildVersionCodes.M)
+			{
+				return GeolocationAccessStatus.Allowed;
+			}
+
+			// check if permission is already granted
+			if (Android.Support.V4.Content.ContextCompat.CheckSelfPermission(Uno.UI.ContextHelper.Current, Android.Manifest.Permission.AccessFineLocation)
```

Comment:
I think we should return `GeolocationAccessStatus.Allowed` if  `Android.Manifest.Permission.AccessFineLocation` or `Android.Manifest.Permission.AccessCoarseLocation` is granted.

Reply:
It is not easy to decide...
But GetGeopositionAsync() emulation is done by using only one option, hard-coded Fine or Coarse. As Coarse can have accuracy over 1 km, I think we should use Fine.
So, we should check Fine, and not Coarse.

Reply:
GetGeoPositionAsync should return a position if either of those permission is requested.
https://developer.android.com/reference/android/location/LocationManager#getLastKnownLocation(java.lang.String)
_Requires Manifest.permission.ACCESS_COARSE_LOCATION or Manifest.permission.ACCESS_FINE_LOCATION_

Reply:
Looking at `InitializeLocationProvider` below it seems it is initializing with Coarse accuracy.

Reply:
Ok, right. I changed it in my much reworked Geolocator, but didn't propagate this change to PR.
Coarse has accuracy about 1 km, so it is not something that is expected by 'windows programmer'.
InitializeLocationProvider  method has unused parameter...

I think I should propagate much more changes to this PR...